packages/app/src/app/pages/Sandbox/Editor/Content/Tabs/ModuleTab/ModuleTab.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import EntryIcons from 'app/pages/Sandbox/Editor/Workspace/Files/DirectoryEntry/Entry/EntryIcons';
+// eslint-disable-next-line import/extensions
+import getType from 'app/utils/get-type.ts';
+import { StyledNotSyncedIcon } from './elements';
+import { TabTitle, TabDir, StyledCloseIcon } from '../Tab/elements';
+import TabContainer from '../TabContainer';
+
+export const ModuleTab = ({
+  dirName,
+  hasError,
+  isNotSynced,
+  tabCount,
+  module,
+  markNotDirty,
+  setCurrentModule,
+  discardModuleChanges,
+  ...props
+}) => {
+  const renderTabStatus = (hovering, closeTab) => {
```

Comment:
@Saeris This can be simplified to the following, which will make it easier to add extra props or new conditions:

```jsx
const getTabStatusProps = (hovering, closeTab) => {
  if (hovering && isNotSynced && tabCount === 1) {
    return { show: 'true' };
  }
  if (hovering && isNotSynced && tabCount > 1) {
    return { onClick: closeTab, show: 'true' };
  }
  if (hovering && tabCount === 1) {
    return { onClick: closeTab };
  }
  if (hovering && tabCount > 1) {
    return { onClick: closeTab, show: 'true' };
  }
  if (!hovering && isNotSynced) {
    return { show: 'true' };
  }

  return {};
};

const renderTabStatus = (hovering, closeTab) => {
  const props = getTabStatusProps(hovering, closeTab);

  return <StyledNotSyncedIcon {...props} />;
};
```

Reply:
Good suggestion. I was in the middle of implementing it then noticed that one of the imports wasn't being used. We're actually using this method to decide which of two components to render in addition to the props on them, so I'm going to keep this mostly as is.